asv/util.py

```diff
@@ -392,7 +394,7 @@ def watcher_run():
                 time.sleep(0.1)
                 if time.time() - start_time[0] > timeout:
                     was_timeout[0] = True
-                    proc.terminate()
+                    proc.send_signal(signal.CTRL_BREAK_EVENT)
```

Comment:
Processes *can* ignore `CTRL_BREAK_EVENT` by registering a handler for `SIGBREAK` (albeit -- at this moment -- this isn't documented neither on [MSDN](https://msdn.microsoft.com/en-us/library/xdkz3x12.aspx#Anchor_2) nor for [Python](https://docs.python.org/3.6/library/signal.html#signal.signal)).

However, as far as I can tell, there isn't much more that we could do:
* [`CTRL_CLOSE_EVENT`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms683242(v=vs.85).aspx#CTRL_CLOSE_EVENT) would be more like `SIGTERM` followed by `SIGKILL` (forced termination after a delay of a few seconds), but there doesn't seem to be a way to trigger it programmaticaly;
* [`TerminateProcess`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms686714(v=vs.85).aspx) is a nearly instant (possibly even overly brutal) way to end a process, but the only way to enumerate processes in a group seems to be through a [snapshot](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682489(v=vs.85).aspx).

Moreover, termination can always be escaped by creating a new group (also under Linux).

I've been thinking about adding a note under [benchmark attributes](http://asv.readthedocs.io/en/latest/writing_benchmarks.html#benchmark-attributes), something like:

> We do our best to terminate the whole benchmark on timeout, but if you create process groups within the benchmark, trap `SIGBREAK` or handle `CTRL_BREAK_EVENT` under Windows, your subprocesses may survive.

Or would this be too detailed?